### ISSUE TITLE
[Bug]: Accessing a link with an invalid hash as a selector will cause the page to get stuck.

### ISSUE THREAD

**Issue by aiji42** [remix-run/remix#1304]
### Which Remix packages are impacted?

- [X] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [ ] `@remix-run/dev`
- [ ] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [X] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

1.1.1

### What version of Node are you using? Minimum supported version is 14.

16.13.0

### Steps to Reproduce

Access with a hash value that is not valid as a selector.

For example, supabase's Magic Link.
`http://127.0.0.1:8787/#access_token=xxxxxxxxxxxx.yyyyyyyyyyy.zzzzzzzzzzzzz&expires_in=3600&refresh_token=aaaaaaaaa&token_type=bearer&type=magiclink`

### Expected Behavior

Access without errors.

### Actual Behavior

The browser console log will output an infinite number of errors and the page gets stuck.
![スクリーンショット 2022-01-01 午後3 39 09](https://user-images.githubusercontent.com/6711766/147845308-03bf4248-4db5-4091-9b67-e4b40ce88d49.png)

**Comment by aiji42** [comment]
The problem is with `ScrollRestoration`, which seems to be caused by the `document.querySelector` on line 102. 
https://github.com/remix-run/remix/blob/1fd70960e4d88740df5bf407a6ba2cd2b9549459/packages/remix-react/scroll-restoration.tsx#L102
try... .catch to fix the problem.

```js
      // try to scroll to the hash
      if (location.hash) {
        let el = null
        try {
          el = document.querySelector(location.hash);
        } catch (e) {
          conosle.warn(e)
        }
        if (el) {
          el.scrollIntoView();
          return;
        }
      }
```

**Comment by na2hiro** [comment]
This seems to be fixed by switching to `getElementById` https://github.com/remix-run/remix/pull/936

**Comment by machour** [comment]
Fixed in #1862
